### PR TITLE
cleaning up indoor positioning related code

### PR DESCRIPTION
**Pull request by marvinmirtschin**

### CODE REVIEW

File: data/phyphox/full recordings/Chlotti_long/annotations.tsv [hunk 1]
```diff
@@ -0,0 +1 @@
+action		00:00:48.700	48.7	00:00:51.800	51.8	00:00:03.100	3.1	pulling  door
```
Comment (marvinmirtschin) [inline data/phyphox/full recordings/Chlotti_long/annotations.tsv:1]: Placeholder to let code run until reading.

File: data/phyphox/full recordings/Chlotti_long/hands.tsv [hunk 2]
```diff
@@ -0,0 +1 @@
+hand		00:00:48.700	48.7	00:00:51.800	51.8	00:00:03.100	3.1	right
```
Comment (marvinmirtschin) [inline data/phyphox/full recordings/Chlotti_long/hands.tsv:1]: Placeholder to let code run until reading.

File: src/indoor_positioning/_calculation.py [hunk 3]
```diff
@@ -3,178 +3,82 @@
...
-    df_new["timestamp"] = timestamps
-    df_new["major"] = [major] * len(timestamps)
-    df_new["minor"] = [minor] * len(timestamps)
-    df_new.drop(columns=["indices"], inplace=True)
-    return df_new
+    missing_timestamps = [timestamp for timestamp in all_timestamps if timestamp not in current_timestamps]
+    missing_values = [missing_value] * len(missing_timestamps)
 
+    missing_frame = {'timestamp': missing_timestamps, 'rssi': missing_values}
+    missing_df = pd.DataFrame(missing_frame)
```
Comment (JuliusCosmoRomeo) [inline src/indoor_positioning/_calculation.py:74]: can we have an assertion here that we have no gaps in the `result` dataframe?

File: src/indoor_positioning/_calculation.py [hunk 4]
```diff
@@ -3,178 +3,82 @@
...
+        df["minor"].replace(11, 5, inplace=True)
+        df["minor"].replace(9, 6, inplace=True)
+    else:
+        # filter out configuration beacon
+        df = df[df["minor"] != 9]
+    return df
 
 
-def batch_data(df: pd.DataFrame, start_timestamp=None, duration=500, aggregation_function="mean"):
+def batch_data(df: pd.DataFrame, start_timestamp=None, end_timestamp=None, duration=500, aggregation_function="mean", missing_value=MISSING_VALUE):
```
[the remark this answers is not in the document]
Comment (JuliusCosmoRomeo) [reply src/indoor_positioning/_calculation.py:45]: also, what is the duration param?

File: src/features/_timeseries_feature_extraction.py [hunk 5]
```diff
@@ -3,63 +3,26 @@
...
-            indoor_features = extract_indoor_feature(timeseries, column_id=['action_id', "segment_id"])
-            timeseries.drop(["rssi", "minor"], axis=1, inplace=True)
-    timeseries.drop(["action_id", "segment_id"], axis=1, inplace=True)
+        indoor_df = timeseries.loc[:, BEACON_MINORS + ["action_id", "segment_id"]]
+        indoor_features = indoor_df.groupby(["action_id", "segment_id"]).apply(np.mean)
+        indoor_features.drop(["action_id", "segment_id"], axis=1, inplace=True)
+
+        if not use_fingerprinting_approach:
+            # use only strongest beacon -> base for maybe using labels later on
+            indoor_features = pd.DataFrame(indoor_features.idxmax(axis=1), columns=["minor"])
```
Comment (JuliusCosmoRomeo) [inline src/features/_timeseries_feature_extraction.py:19]: nice, I like that solution now! 👍

Comment (JuliusCosmoRomeo) [reply src/features/_timeseries_feature_extraction.py:19]: can we anyways add 2 assertions here? One for each implementation of the indoor localization to check that the feature columns are correct.

**Pull request merged**